examples/friend-list/src/domain/services/search.ts

```diff
@@ -0,0 +1,12 @@
+import { filter, toLowerCase } from '167'
+
+import { Friends } from '../model'
+
+const includes = (query: string = '') => (value: string = '') =>
+  toLowerCase(value).includes(toLowerCase(query))
+
+export function search(query: string, friends: Friends): Friends {
```

Comment:
This is not a domain service because it doesn’t _create_ an entity or aggregate. Domain services are used purely to construct (complex) entities or aggregates. This is an application service.